packages/url/init.lua

```diff
@@ -147,6 +153,12 @@ function package:registerCommands ()
         secondaryPenalty = secondaryPenalty,
         worsePenalty = worsePenalty
       })
+      local brackets = SILE.settings:get("url.brackets")
+      brackets = SU.splitUtf8(brackets)
+      if #brackets == 2 then
+        result[1] = brackets[1] .. result[1]
+        result[#result] = result[#result] .. brackets[2]
```

Comment:
This is wrong / not robust, breaking current behavior:
It's far too bold to assume that `result[#result]` is a string. Here it can be a penalty, since this is invoked _after_ the inputfilter has been applied. You do need to add a test scenario here, to check what happens if the URL ends with a "/" or other characters after which an extra penalty might have been inserted... I did a quick check and it oops:
```
error summary:
	Processing at: experiments/url.sil:4:5: in \url
	Using code at: ./packages/url/init.lua:150: attempt to concatenate a table value (field '?')
```
So `result[#result+1] = bracket[2]` would be the solution here...

But that could be true to for the first element, it a penalty was inserted _before_ the actual text. Currently it only occurs for URL fragments, so it's unlikely at start position, but for the sake of robustness you should probably handle it too... 

... Because the content provided by the user might not even be a string! E.g. all of the following URL currently works:

```
\url{\img[src=google.png] http://google.com/}
\url{\strong{https:}//google.com/}
\url{\underline{https://google.com}}
```
Whether the above is fully legit or not, it does work as of today... And breaks when you start to consider the content is a string.

Reply:
Thank you. I wasn't too sure about this one. I appreciate the thorough review. Your comment above seems a better solution than mine and I'll implement it. :+1:

Reply:
(The `urlstyle` comment in case not clear.)

Reply:
You are welcome. I actually stumbled on the `\code` stuff in the url package in #1194 (that's one of the unaddressed tick box at this date), but I didn't knew as much at the time. Would it be now, by the way, I'd propose to consider moving the default definition of "code" to the plain class (besides "strong" and "em") - it's used in plenty of places (manual, package docs, markdown, etc.) and while assuming the url package to be generally loaded is quite reasonable for most scenarios, it still does feel out of place, IMHO.

Reply:
I agree `\code` is out of place in the *url* package and should probably be moved to the *plain* class.

A minor side thought is that we don't currently request a monospace font be installed by packages. We do request Gentium Plus as default and we also suggest Libertinus Math (math package default) and Noto Sans CJK JP (*jplain*/*jbook* class default) but we don't similarly suggest Hack (verbatim package default).

A default mono font family should probably be a top level setting and that wrapper function in verbatim should use the setting instead of setting it directly.